packages/transform/src/lib/TransformMiddleware.ts

```diff
@@ -0,0 +1,246 @@
+import { ApplyMiddlewareOptions, JoshMiddleware, Method, Payloads, PostProvider, PreProvider } from '@joshdb/provider';
+import type { Primitive } from '@sapphire/utilities';
+
+@ApplyMiddlewareOptions({ name: 'transform' })
+export class TransformMiddleware<StoredValue = unknown> extends JoshMiddleware<TransformMiddleware.ContextData<StoredValue>, StoredValue> {
+  @PreProvider()
+  public async [Method.Set]<Value = StoredValue>(payload: Payloads.Set<Value>): Promise<Payloads.Set<Value>> {
+    const { key, path } = payload;
+    const { before } = this.context;
+
+    await this.provider[Method.Set]({
+      method: Method.Set,
+      errors: [],
+      key,
+      path,
+      value: await before(payload.value as unknown as StoredValue, null, null)
+    });
+
+    return payload;
+  }
+
+  @PreProvider()
+  public async [Method.SetMany](payload: Payloads.SetMany): Promise<Payloads.SetMany> {
+    const { entries } = payload;
+    const { before } = this.context;
+
+    for (const { key, value, path } of entries) {
+      await this.provider[Method.Set]({
+        method: Method.Set,
+        errors: [],
+        key,
+        path: [],
+        value: await before(value as StoredValue, key, path)
+      });
+    }
+
+    return payload;
+  }
+
+  @PreProvider()
+  public async [Method.Update]<Value = StoredValue>(payload: Payloads.Update<StoredValue, Value>): Promise<Payloads.Update<StoredValue, Value>> {
+    const { key, hook } = payload;
+    const { before, after } = this.context;
+
+    await this.provider[Method.Update]({
+      method: Method.Update,
+      errors: [],
+      key,
+      hook: async (value) => {
+        const result = await hook(await after(value, key, []), key);
+
+        return before(result as StoredValue, key, []);
+      }
+    });
+
+    return payload;
+  }
+  // POST PROVIDER
+
+  @PostProvider()
+  public async [Method.Entries]<Value = StoredValue>(payload: Payloads.Entries<Value>): Promise<Payloads.Entries<Value>> {
+    payload.data ??= {};
+
+    const { data } = await this.provider[Method.Entries]({ method: Method.Entries, errors: [] });
+    const { after } = this.context;
+
+    for (const [key, value] of Object.entries(data!)) {
+      payload.data[key] = (await after(value as StoredValue, key, null)) as Value;
+    }
+
+    return payload;
+  }
+
+  public async [Method.Every](payload: Payloads.Every.ByHook<StoredValue>): Promise<Payloads.Every.ByHook<StoredValue>>;
+  public async [Method.Every](payload: Payloads.Every.ByValue): Promise<Payloads.Every.ByValue>;
+  @PostProvider()
+  public async [Method.Every](payload: Payloads.Every<StoredValue>): Promise<Payloads.Every<StoredValue>> {
+    const { type, hook, path, value } = payload;
+    const { data } = await this.provider[Method.Every]({ method: Method.Every, errors: [], type, hook, path, value });
```

Comment:
Why?

Reply:
I generally have no clue, it seems really pointless now.